pandas/core/frame.py

```diff
@@ -4989,8 +4989,9 @@ def sort_index(
             and 1 identifies the columns.
         level : int or level name or list of ints or list of level names
             If not None, sort on values in specified index level(s).
-        ascending : bool, default True
-            Sort ascending vs. descending.
+        ascending : bool or list of bools, default True
```

Comment:
Can you update the annotation for this function to reflect this?

Reply:
So originally I had given it a Union type but mypy complained because here the ascending argument seemingly has to be a singleton bool: https://github.com/pandas-dev/pandas/blob/master/pandas/core/series.py#L3138. I guess this is unavoidable because it has no way of knowing that it shouldn't get here when ascending is a list?

Reply:
Wouldn't that conflict with the documentation updates?

Reply:
Hmm, I think it's okay since it "can" accept a list of bools, just conditional on the index being a MultiIndex?

Reply:
Not sure - looks like this works (probably shouldn't)

```python
>>> pd.DataFrame([[1]]).sort_index(ascending=[True, False, True])
   0
0  1
```